app/models/paper.rb

```diff
@@ -1,10 +1,31 @@
 class Paper < ApplicationRecord
   acts_as_taggable
+  acts_as_taggable_on :biases, :methodologies
+  ActsAsTaggableOn.strict_case_match = true
 
   has_and_belongs_to_many :authors
   has_many :lists, through: :references
   has_many :references, dependent: :destroy
 
   accepts_nested_attributes_for :authors, reject_if: proc { |attributes| attributes['name'].blank? }
   validates_associated :authors
+  validate :allowed_biases, :allowed_methodologies
+
+  def allowed_biases
+	  invalid_biases = bias_list - valid_biases
+	  invalid_biases.each {|b| errors.add(b, 'is not an accepted bias') }
+	end
+
+  def valid_biases
+    %w(selection performance detection attrition reporting small\ sample\ size)
```

Comment:
ah ok

On Wed, Aug 10, 2016 at 12:10 AM, Brian Kung <notifications@github.com>
wrote:

> In app/models/paper.rb
> <https://github.com/briankung/projectcredo/pull/49#discussion_r74186271>:
>
> >
> >    has_and_belongs_to_many :authors
> >    has_many :lists, through: :references
> >    has_many :references, dependent: :destroy
> >
> >    accepts_nested_attributes_for :authors, reject_if: proc { |attributes| attributes['name'].blank? }
> >    validates_associated :authors
> > +  validate :allowed_biases, :allowed_methodologies
> > +
> > +  def allowed_biases
> > +	  invalid_biases = bias_list - valid_biases
> > +	  invalid_biases.each {|b| errors.add(b, 'is not an accepted bias') }
> > +	end
> > +
> > +  def valid_biases
> > +    %w(selection performance detection attrition reporting small\ sample\ size)
>
> I mean, put them next to each other.
>
> def valid_biases
>   # ...end
> def valid_methodologies
>   # ...end
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/briankung/projectcredo/pull/49/files/ddc284b119a291513400a37b5e5a8c2366d63154#r74186271>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACKsZBug3c6Bfig5HZEf1db0xFTRVakNks5qeV1hgaJpZM4JflY4>
> .
>